giskard/scanner/scanner.py

```diff
@@ -183,6 +148,64 @@ def _postprocess(self, issues: Sequence[Issue]) -> Sequence[Issue]:
 
         return issues
 
+    def _validate_model_and_dataset(self, model, dataset):
+        if not isinstance(model, BaseModel):
+            raise ValueError(
+                "The model object you provided is not valid. Please wrap it with the `giskard.Model` class. "
+                "See the instructions here: https://docs.giskard.ai/en/latest/guides/wrap_model/index.html"
+            )
+
+        if dataset is not None and not isinstance(dataset, Dataset):
+            raise ValueError(
+                "The dataset object you provided is not valid. Please wrap your dataframe with `giskard.Dataset`. "
+                "You can follow the docs here: https://docs.giskard.ai/en/latest/guides/wrap_dataset/index.html"
+            )
```

Comment:
let's move this to `model_validation` and `dataset_validation`

Reply:
This belongs to the scan, it checks that the arguments coming in are instances of the right objects.

Reply:
yeah well that's what the validation step does